tools/version.sh

```diff
@@ -112,7 +112,7 @@ if [ "X${MAJOR}.${MINOR}" = "X${VERSION}" ]; then
   echo $ADVICE
   exit 4
 fi
-PATCH=`echo ${VERSION} | cut -d'.' -f3`
+PATCH=`echo ${VERSION} | cut -d'-' -f1 | cut -d'.' -f3`
```

Comment:
Can we support symbol other than '-'?

Reply:
Well, it is not usual, we usually use '-' for delimiter the extra version. I tested it with other cases, let's say if we have a tag that doesn't have the expected delimiter it will include the remaining of the tag in the PATCH

Reply:
PATCH expect a number noramlly, can we extract the character util a non-number is found?

Reply:
@xiaoxiang781216 thanks for the review. I updated the commit, now it's just adding the numbers, whatever comes after the numbers are out of the PATCH variable.

Reply:
The change looks good, thanks.